### PR TITLE
[develop2] test for relocatable toolchains

### PR DESCRIPTION
**Pull request by memsharded**
Changelog: omit
Docs: omit

Close https://github.com/conan-io/conan/issues/5059

This test allows:
- To create relocatable toolchain packages
- A relocatable toolchain package requires some customization in every machine, that would break immutability
- We an use a base immutable package with the heavy bits, that take long to build
- Another package ``tool_requires`` this one, and uses ``build_policy="missing"`` and ``upload_policy="skip"`` to avoid uploading, and to build it when necessary, in a transparent way
- This other package implements in its ``package()`` or ``build()`` or both the necessary customizations to the current machine, copying from the other package what it needs

### CODE REVIEW

File: conan/cli/printers/graph.py [hunk 1]
```diff
@@ -99,7 +99,7 @@ def _format_requires(title, reqs_to_print):
             return
         output.info(title, Color.BRIGHT_YELLOW)
         for pref, (status, remote) in sorted(reqs_to_print.items(), key=repr):
-            if remote is not None:
+            if remote is not None and status != BINARY_SKIP:
```
Comment (franramirez688) [inline conan/cli/printers/graph.py:99]: Is this a fix related to the PR?

Comment (memsharded) [reply conan/cli/printers/graph.py:99]: yes, otherwise the checks ``c.assert_listed_binary({"base/1.0": ("62e589af96a19807968167026d906e63ed4de1f5", "Skip"),`` failed as it was printing ``Skip (default)`` which doesn't make much sense

File: conans/test/integration/build_requires/test_relocatable_toolchain.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
+import textwrap
+
+from conans.test.assets.genconanfile import GenConanfile
+from conans.test.utils.tools import TestClient
+
+
+def test_relocatable_toolchain():
```
Comment (franramirez688) [inline conans/test/integration/build_requires/test_relocatable_toolchain.py:7]: A docstring explaining the test and a link to the issue that closes would be great.

**Review by franramirez688 — APPROVED** [review]

**Review by czoido — APPROVED** [review]

**Review by AbrilRBS — APPROVED** [review]
Also TIL that you can specify `upload_policy`!

File: conans/test/integration/build_requires/test_relocatable_toolchain.py (hunk 2 above)
Comment (memsharded) [reply conans/test/integration/build_requires/test_relocatable_toolchain.py:7]: Done
